### ISSUE TITLE
IDEA:  Combing Feature to Stop Stringing

### ISSUE THREAD

**Issue by zChas** [alexrj/Slic3r#4685]
Problem:  Stringing in some prints

Feature to help reduce/stop stringing:  "Combing"  Option

Details:  see YouTube vid "Stop Ender 3 Stringing with this One Weird Trick!"  https://youtu.be/_QRb54zVPfQ?t=5m48s

 ...  also:  https://ultimaker.com/en/resources/52507-travel

**Comment by lordofhyphens** [comment]
Could you elaborate/describe combing and how it may differ from "Avoid Crossing Perimeters"?

**Comment by supermerill** [comment]
Seems like a wipe that go "inside" the part.

**Comment by lordofhyphens** [comment]
Wipe is already implemented in Printer settings, though, for retraction (pulls it into the infill first). 

I've thought about enabling "wipe" move generations when firmware retract is implemented (to get some of the benefits and still have the simpler HW retract), though.

**Comment by sfjuocekr** [comment]
If you print a threaded pipe, Slic3r travels from one side to the other with "Avoid Crossing Perimeters" enabled and causes some stringing between travels. 

Cura with combing mode enabled does the same, but does not introduce stringing.

This is the only way I can "enforce" stringing to happen though, so I I'd blame the cheap filament I'm using but Cura does a much better job in this edge case ;)

Basically what happens in Cura is that it moves inside the model before movement and these strings are eliminated:

![IMG_20190613_012539](https://user-images.githubusercontent.com/398018/59393059-3c609580-8d7a-11e9-9471-e67449252881.jpg)

**Comment by Noozle3D** [comment]
I would also like a feature like "Combing" for Slic3r PE. This is the only reason I don't currently use Slic3r. I work mostly with TPU and this stuff constantly oozes. Nozzle wipe is just not enough.

I have attached screen captures of "Avoid Crossing Perimeters" ON and OFF and Curas' "Combing". "Avoid Crossing Perimeters" seems to reduce the nozzle from leaving the print about half as much, but "Coming" reduces this almost completely (the travels visible are due to islands on the print).
Not sure if "Avoid Crossing Perimeters" is broken or is intended for a different reason.

![Avoid Crossing Perimeters OFF](https://user-images.githubusercontent.com/55472137/65103480-5ecb0600-d99c-11e9-8fdf-53d7338f825c.PNG)
![Avoid Crossing Perimeters ON](https://user-images.githubusercontent.com/55472137/65103481-5ecb0600-d99c-11e9-8270-d736e71e8539.PNG)
![Combing](https://user-images.githubusercontent.com/55472137/65103971-da798280-d99d-11e9-808d-d4b9647b9241.PNG)

**Comment by lordofhyphens** [comment]
More features desired than people available to work on them.

**Comment by klack** [comment]
I have also seen great results using Cura's "Combing" feature and came here looking for it in slic3r.  This would allow me to completely move to slic3r.

**Comment by edslatt** [comment]
+1

I would love to have the "Combing" feature.

**Comment by tinuva** [comment]
+!
